constraints.pro

```diff
@@ -0,0 +1,152 @@
+%%% Utility predicates
+
+is_valid_version_range(VersionRange) :-
```

Comment:
This likely doesn't cover all uses cases (for instance, what if want to use a GitHub repo reference instead of a version range), but we can cross that bridge later.